Libadwaita Leaflet

Hello, I just updated to the 0.5.0 beta and adjusted my components,
but I'm having trouble getting libadwaita's `Leaflet` to work.

```
error[E0599]: the method `set_child` exists for struct `Leaflet`, but its trait bounds were not satisfied
   --> src/main.rs:128:17
    |
128 |                   set_child = Box {
    |                   ^^^^^^^^^
```

```
4  | / glib::wrapper! {
25  | |     #[doc(alias = "AdwLeaflet")]
26  | |     pub struct Leaflet(Object<ffi::AdwLeaflet, ffi::AdwLeafletClass>) @extends gtk::Widget, @implements gtk::Accessible, gtk::Buildable, ...
27  | |
...   |
30  | |     }
31  | | }
    | | -
    | | |
    | | doesn't satisfy `Leaflet: IsA<Bin>`
    | | doesn't satisfy `Leaflet: IsA<FlowBoxChild>`
    | | doesn't satisfy `Leaflet: IsA<Frame>`
    | | doesn't satisfy `Leaflet: IsA<ListBoxRow>`
    | | doesn't satisfy `Leaflet: IsA<Popover>`
    | | doesn't satisfy `Leaflet: IsA<gtk4::Button>`
    | | doesn't satisfy `Leaflet: IsA<gtk4::ComboBox>`
    | | doesn't satisfy `Leaflet: IsA<gtk4::Window>`
    | | doesn't satisfy `Leaflet: gtk4::prelude::ButtonExt`
    | | doesn't satisfy `Leaflet: gtk4::prelude::ComboBoxExt`
    | | doesn't satisfy `Leaflet: gtk4::prelude::FlowBoxChildExt`
    | | doesn't satisfy `Leaflet: gtk4::prelude::FrameExt`
    | | doesn't satisfy `Leaflet: gtk4::prelude::GtkWindowExt`
    | | doesn't satisfy `Leaflet: gtk4::prelude::ListBoxRowExt`
    | |_doesn't satisfy `Leaflet: gtk4::prelude::PopoverExt`
    |   doesn't satisfy `Leaflet: libadwaita::prelude::BinExt`
    |
    = note: the following trait bounds were not satisfied:
            `Leaflet: IsA<gtk4::Button>`
            which is required by `Leaflet: gtk4::prelude::ButtonExt`
            `Leaflet: IsA<gtk4::ComboBox>`
            which is required by `Leaflet: gtk4::prelude::ComboBoxExt`
            `Leaflet: IsA<FlowBoxChild>`
            which is required by `Leaflet: gtk4::prelude::FlowBoxChildExt`
            `Leaflet: IsA<Frame>`
            which is required by `Leaflet: gtk4::prelude::FrameExt`
            `Leaflet: IsA<ListBoxRow>`
            which is required by `Leaflet: gtk4::prelude::ListBoxRowExt`
            `Leaflet: IsA<Popover>`
            which is required by `Leaflet: gtk4::prelude::PopoverExt`
            `Leaflet: IsA<gtk4::Window>`
            which is required by `Leaflet: gtk4::prelude::GtkWindowExt`
            `Leaflet: IsA<Bin>`
            which is required by `Leaflet: libadwaita::prelude::BinExt`

error[E0277]: the trait bound `Leaflet: RelmSetChildExt` is not satisfied
   --> src/main.rs:127:13
    |
127 |             Leaflet {
    |             ^^^^^^^ the trait `RelmSetChildExt` is not implemented for `Leaflet`
    |
    = help: the following other types implement trait `RelmSetChildExt`:
              Bin
              Clamp
              ClampScrollable
              FlowBoxChild
              Frame
              LinkButton
              ListBoxRow
              Popover
            and 12 others
    = note: required because of the requirements on the impl of `RelmContainerExt` for `Leaflet`
```

I also tried this without the `set_child`, in which case, 
only the second message is being shown.

I'd assume that I can implement the missing trait myself, 
but I think it should be included by default, 
when using the `libadwaita` feature.

I've tried forking this repo and adding `adw::Leaflet` 
to the `impl`-ement macros in the `relm4/src/extensions` folder, 
which seems to be the correct way.

Unfortunately, 
this does not appear to be enough,
as it's now throwing an error, which seems to originate from the gtk4 library itself:
```
error[E0599]: the method `set_child` exists for reference `&Leaflet`, but its trait bounds were not satisfied

   |
15 |                       self.set_child(widget.map(|w| w.as_ref()));
   |                            ^^^^^^^^^
...
56 | /     set_child_impl!(
57 | |         adw::Bin,
58 | |         adw::Clamp,
59 | |         adw::ClampScrollable,
...  |
62 | |         adw::Leaflet
63 | |     );
   | |_____- in this macro invocation


24 | / glib::wrapper! {
25 | |     #[doc(alias = "AdwLeaflet")]
26 | |     pub struct Leaflet(Object<ffi::AdwLeaflet, ffi::AdwLeafletClass>) @extends gtk::Widget, @implements gtk::Accessible, gtk::Buildable, ...
27 | |
...  |
30 | |     }
31 | | }
   | | -
   | | |
   | |_doesn't satisfy `Leaflet: BinExt`
   |   doesn't satisfy `Leaflet: gtk4::prelude::IsA<Bin>`
   |
   = note: the following trait bounds were not satisfied:
           `Leaflet: gtk4::prelude::IsA<Bin>`
           which is required by `Leaflet: BinExt`
           `&Leaflet: gtk4::prelude::IsA<Bin>`
           which is required by `&Leaflet: BinExt`
   = note: this error originates in the macro `set_child_impl` (in Nightly builds, run with -Z macro-backtrace for more info)
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,6 +2,7 @@
 
 ## Unreleased
 
++ core: Implement RelmContainerExt for Leaflet, Carousel and TabView
 + relm4-components: Port `OpenButton` to 0.5
 + core: Add `iter()` method to `FactoryVecDeque`
 + core: Rename `ParentMsg` and `output_to_parent_msg` to `ParentInput` and `output_to_parent_input`, respectively.

--- a/relm4/src/extensions/container.rs
+++ b/relm4/src/extensions/container.rs
@@ -40,3 +40,6 @@ macro_rules! add_child_impl {
 
 append_impl!(gtk::Box, gtk::ListBox);
 add_child_impl!(gtk::InfoBar, gtk::Stack);
+
+#[cfg(feature = "libadwaita")]
+append_impl!(adw::Leaflet, adw::Carousel, adw::TabView);

